### ISSUE TITLE
Add missing statistical reducers

### ISSUE THREAD

**Issue by oleksiyskononenko** [h2oai/datatable#2999]
RIght now the following reducers are only available as the corresponding `Frame` methods, but can't be used in the square brackets syntax as `dt.*()`:

- [ ] `mode()`
- [ ] `nmodal()`
- [ ] `nunique()`
- [ ] `countna()`

**Comment by pradkrish** [comment]
I accidentally converted the last task into an issue and then closed immediately realising the mistake. It is still linked here somehow, anyway to remove it? Sorry for the inconvenience.

**Comment by st-pasha** [comment]
That's ok. Looks like it's a new feature they added only a few days ago. Good to know!

**Comment by pradkrish** [comment]
Should I link the PR for `countna` to this issue or shall I reopen the closed issue and link it to that one?

**Comment by oleksiyskononenko** [comment]
@pradkrish Yeah, seems to be a new feature. I've restored this issue to its original state, in this case I guess it is better to keep track of the missing reducers from one issue rather than distributing it into four smaller ones. For other issues, we may want to use the feature you just discovered :) 

On your PR just replace "Related" with "WIP for".
